src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java

```diff
@@ -59,4 +62,45 @@ public ArrayList<Course> getSectionsByPsId(@Parameter(name = "psId") @RequestPar
     }
     return sections;
   }
+
+  @Operation(summary = "Delete a schedule and associated lectures by enroll code and psId")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @DeleteMapping(value = "/delete")
+  public Object deleteScheduleAndLectures(
+      @Parameter(name = "psId") @RequestParam Long psId,
+      @Parameter(name = "enrollCd") @RequestParam String enrollCd)
+      throws JsonProcessingException {
+    User currentUser = getCurrentUser().getUser();
+
+    PersonalSchedule ps =
+        personalScheduleRepository
+            .findByIdAndUser(psId, currentUser)
+            .orElseThrow(() -> new EntityNotFoundException(PersonalSchedule.class, psId));
+    Iterable<PSCourse> courses = coursesRepository.findAllByPsId(psId);
+    boolean courseFound = false;
+    ArrayList<String> relatedEnrollCodes = new ArrayList<>();
+    String body = ucsbCurriculumService.getAllSections(enrollCd, ps.getQuarter());
+    if (!body.equals("{\"error\": \"401: Unauthorized\"}")
+        && !body.equals("{\"error\": \"Enroll code doesn't exist in that quarter.\"}")) {
+      Iterator<JsonNode> it = objectMapper.readTree(body).path("classSections").elements();
+      while (it.hasNext()) {
+        JsonNode classSection = it.next();
+        String sectionEnrollCd = classSection.path("enrollCode").asText();
+        relatedEnrollCodes.add(sectionEnrollCd);
+      }
+    }
+    for (PSCourse crs : courses) {
+      if (relatedEnrollCodes.contains(crs.getEnrollCd())) {
+        courseFound = true;
+        coursesRepository.delete(crs);
+      }
+    }
+    if (!courseFound) {
+      throw new EntityNotFoundException(
+          PSCourse.class, "enrollCd: " + enrollCd + " and psId: " + psId);
+    }
+    return genericMessage(
+        "Schedule with psId %s and associated lectures with enrollCd %s deleted"
+            .formatted(psId, enrollCd));
+  }
```

Comment:
Can we move this code to the existing PSCourseController.java instead of putting it here?

And let's add another story to move the other api endpoint over into PSCourseController as well.  This controller should probably not have been built in the first place as a separate file.

Reply:
Ok, fine, since we are close to the deadline, I'll let it go.. but I hate the fact that we have two controller files and two controller tests where we should have one.